### PR TITLE
Add "Hello World" example to basic init

### PR DESCRIPTION
**Pull request by JamesJeffryes**
OK, this was more painful than expected but you should be able to initialize Python, Perl and Java modules and run `kb-sdk test` to produce a hello world report.

### CODE REVIEW

**Comment by psdehal** [comment]
so the behavior now would be that kb-sdk init creates a run_module function
in the impl file that prints "hello world"

**Comment by JamesJeffryes** [comment]
Actually the created report says "Hello World" (in the test) or what ever you specify in "parameter_1"

**Comment by jayrbolton** [comment]
Testing and looking through things

File: src/java/us/kbase/mobu/initializer/ModuleInitializer.java [hunk 1]
```diff
@@ -175,69 +175,61 @@ public void initialize(boolean example) throws Exception {
...
-						false, // core or sync clients
-						false, // dynamic client
-						"release", //tagVer
-						this.verbose,
-						dependantModuleName,
-						null,
-						null // clientName
-					);
-			}
+            new ClientInstaller(new File(moduleDir), false).install(
```
Comment (jayrbolton) [inline src/java/us/kbase/mobu/initializer/ModuleInitializer.java]: lol these params are wild

File: src/java/us/kbase/templates/module_test_java_client.vm.properties [hunk 2]
```diff
@@ -189,6 +192,10 @@ public class ${java_module_name}ServerTest {
         // Check returned data with
         // Assert.assertEquals(..., ret.getSomeProperty());
         // ... or other JUnit methods.
+        Map<String, UObject> params = new HashMap<>();
+        params.put("workspace_name", new UObject(getWsName()));
+        params.put("parameter_1", new UObject("Hello World"));
+        final ReportResults ret = impl.runJavafoo(params, token, getContext());
```
Comment (jayrbolton) [inline src/java/us/kbase/templates/module_test_java_client.vm.properties]: I think this should be something like `impl.runTestJavaHelloWorld(...`

The java tests don't pass for me as it is now.

Comment (jayrbolton) [reply src/java/us/kbase/templates/module_test_java_client.vm.properties]: Changing this to `impl.runTestJavaHelloWorld` seems to fix tests

**Comment by jayrbolton** [comment]
There was an issue with the Java test that I commented about above. The perl version seems to work fine (although I can't make heads or tails of that test file, wowzers). The python version throws an error for me:

```py
Traceback (most recent call last):
  File "/miniconda/lib/python3.6/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/miniconda/lib/python3.6/site-packages/nose/loader.py", line 523, in makeTest
    return self._makeTest(obj, parent)
  File "/miniconda/lib/python3.6/site-packages/nose/loader.py", line 582, in _makeTest
    return MethodTestCase(obj)
  File "/miniconda/lib/python3.6/site-packages/nose/case.py", line 346, in __init__
    self.inst = self.cls()
TypeError: __init__() missing 1 required positional argument: 'config'
```

@JamesJeffryes any ideas? It seems to run the actual test method before the error is raised. I can't tell how or where it gets raised.

**Comment by JamesJeffryes** [comment]
So the Java issue was some bad copy pasta that I missed because I kept using "javafoo" as my module name. I actually can't replicate the Python error you are seeing though. Do you see it when you init with the `-e` flag too?

**Comment by jayrbolton** [comment]
It looks like default tests do work on the filter contigs example, but I still get the above error on the hello world example

**Comment by jayrbolton** [comment]
I finally found the cause of the python error. It's pretty obscure and probably not related to this update.

I was calling my modules "test_python_hello_world_etc". That "test_" prefix on the module name was causing nose2 to do something weird with it. When I rename my module to python_hello_world, it is fixed. Something to note on the troubleshooting page I guess.

**Pull request merged**